### PR TITLE
fix reshape doc, remove reverse api

### PR DESCRIPTION
**Pull request by mapingshuo**
### PR types

Bug fixes
### PR changes

Docs
### Describe

- Fix docs of `paddle.reshape` API
- remove `paddle.reverse` api, since it is duplicated with the `paddle.flip` api.

### CODE REVIEW

**Comment by mapingshuo** [comment]
image
image
image
image

File: python/paddle/tensor/manipulation.py [hunk 1]
```diff
@@ -1309,8 +1303,6 @@ def reshape(x, shape, name=None):
             import numpy as np
             import paddle
 
-            paddle.disable_static()
-
             data = np.random.random([2, 4, 6]).astype("float32")
             x = paddle.to_tensor(data)
```
Comment (jzhang533) [inline python/paddle/tensor/manipulation.py]: 构造数据可以直接用paddle.rand
paddle.fill_constant改为用paddle.full

Comment (mapingshuo) [reply python/paddle/tensor/manipulation.py]: done, 感谢指正。

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by zhiqiu — APPROVED** [review]
LGTM

**Review by jzhang533 — APPROVED** [review]
lgtm

**Pull request merged**